aws_crt_common_ws/aws-crt-c-flags/src/lib.rs

```diff
@@ -0,0 +1,469 @@
+        self.public_defines
+            .push((key.parse().unwrap(), val.parse().unwrap()));
+        self
+    }
+
+    /// Adds a library to the linker line. Formatting for things like framework-vs-system lib are the responsibility
+    /// of the caller. All link targets are transitively passed to module consumers.
+    ///
+    /// # Arguments
+    ///
+    /// * `l_flag` - Linker target. Example: "curl", "Kernel32", "framework=CoreFoundation"
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// build_info.add_link_target("crypto")
+    ///     .add_link_target("framework=Security");
+    /// ```
+    pub fn add_link_target(&mut self, l_flag: &str) -> &mut CRTModuleBuildInfo {
+        self.link_targets.push(l_flag.parse().unwrap());
+        self
+    }
+
+    /// Makes this module build as a shared library. The default is static.
+    pub fn make_shared_lib(&mut self) -> &mut CRTModuleBuildInfo {
+        self.shared_lib = true;
+        self
+    }
+
+    /// Sets the linker search path. Currently this is unused, but if you were to be using this
+    /// module to link against a library built via. something other than cargo, you'd use this to do so.
+    /// The default is the cargo build output directory.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - File system path to where to find libraries to link against (e.g. /usr/lib)
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// use std::path::Path;
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// build_info.set_linker_search_path(Path::new("/opt/where/i/installed/my/manually/built/libcrypto/lib"))
+    ///     .add_link_target("crypto");
+    /// ```
+    pub fn set_linker_search_path(&mut self, path: &Path) -> &mut CRTModuleBuildInfo {
+        self.linker_path = Some(path.to_path_buf());
+        self
+    }
+
+    /// adds an additional include directory to your module build. This is mainly useful only
+    /// if you're using a 3rd party library, not built by this library, and you need the compiler
+    /// to fine the header files.
+    ///
+    /// # Arguments
+    ///
+    /// * `dir` - File system path to where to find the header files you need.
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// use std::path::Path;
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// build_info.add_third_party_include_dir(Path::new("/opt/where/i/installed/my/manually/built/libcrypto/include"))
+    ///     .set_linker_search_path(Path::new("/opt/where/i/installed/my/manually/built/libcrypto/lib"))
+    ///     .add_link_target("crypto");
+    /// ```
+    pub fn add_third_party_include_dir(&mut self, dir: &Path) -> &mut CRTModuleBuildInfo {
```

Comment:
Another way to do this might be to have `add_include_dir` that takes an enum(str) of include directory types (`e.g. External` (third party), `Private`(only needed to build), `Public` (implies copy to output)), which would be more semantically clear